### PR TITLE
eip3860

### PR DESCRIPTION
**Pull request by yann300**
fix https://github.com/ethereum/remix-project/issues/3640

in order to test eip170, compile and deploy https://github.com/yann300/remix-reward (without optimization enabled).

in order to test eip3860, update remix reward to have a unique param for the constructor: `bytes32` compile, copy the bytecode from (the solidity tab), go to udapp, deploy remix reward using the previously copied data.
(note that the copied bytecode can be anything it is really just to have enough data to reach max code size).

### CODE REVIEW

File: apps/remix-ide/src/app/tabs/locales/zh/udapp.json [hunk 1]
```diff
@@ -27,7 +27,7 @@
   "udapp.contractOptionsTitle3": "选择并编译 *.sol 文件以部署或访问合约。",
   "udapp.contractOptionsTitle4": "当有编译的 .sol 文件时，选择 {br} 合约进行部署或与 AtAddress 一起使用。",
   "udapp.checkSumWarning": "您似乎没有使用 checksumed address 。{br} checksumed address 是包含大写字母的地址，如 {a} 中所指定。{br} checksumed address 旨在帮助防止用户将交易发送到错误地址。",
-  "udapp.isOverSizePrompt": "合约创建初始化返回长度超过24576字节的数据。部署可能会失败。 {br}更多信息：{a}",
+  "udapp.isOverSizePromptEip170": "合约创建初始化返回长度超过24576字节的数据。部署可能会失败。 {br}更多信息：{a}",
```
Comment (ioedeveloper) [inline apps/remix-ide/src/app/tabs/locales/zh/udapp.json:30]: You forgot to add translation for `udapp.isOverSizePromptEip3860`

File: libs/remix-ui/run-tab/src/lib/actions/deploy.ts [hunk 2]
```diff
@@ -64,10 +64,19 @@ export const getSelectedContract = (contractName: string, compiler: CompilerAbst
 
       return txHelper.inputParametersDeclarationToString(constructorInteface.inputs)
     },
-    isOverSizeLimit: () => {
-      const deployedBytecode = contract.object.evm.deployedBytecode
+    isOverSizeLimit: async (args: string) => {
+      const encodedParams = await txFormat.encodeParams(args, txHelper.getConstructorInterface(contract.object.abi))
+      const bytecode = contract.object.evm.bytecode.object + (encodedParams as any).dataHex
+      // https://eips.ethereum.org/EIPS/eip-3860
+      const initCodeOversize = (bytecode && bytecode.length / 2 > 2 * 24576)
```
Comment (ioedeveloper) [inline libs/remix-ui/run-tab/src/lib/actions/deploy.ts:71]: Can you separate condition with brackets to make it easier to read?

File: libs/remix-ui/run-tab/src/lib/components/contractDropdownUI.tsx [hunk 3]
```diff
@@ -246,10 +246,19 @@ export function ContractDropdownUI (props: ContractDropdownProps) {
...
       <div>
-        <FormattedMessage id='udapp.isOverSizePrompt' values={{ br: <br />, a: <a href="https://github.com/ethereum/EIPs/blob/master/EIPS/eip-170.md" target="_blank" rel="noreferrer">eip-170</a> }} />
+        { 
+        values.overSizeEip170 && <div>
+          <FormattedMessage id='udapp.isOverSizePromptEip170' values={{ br: <br />, a: <a href="https://eips.ethereum.org/EIPS/eip-170" target="_blank" rel="noreferrer">eip-170</a> }} />
+        </div>
+        }
+        { 
+        values.overSizeEip3860 && <div>
+          <FormattedMessage id='udapp.isOverSizePromptEip3860' values={{ br: <br />, a: <a href="https://eips.ethereum.org/EIPS/eip-3860" target="_blank" rel="noreferrer">eip-170</a> }} />
```
Comment (ioedeveloper) [inline libs/remix-ui/run-tab/src/lib/components/contractDropdownUI.tsx:259]: The text here should be changed to eip-3860

File: libs/remix-ui/run-tab/src/lib/actions/deploy.ts (hunk 2 above)
Comment (yann300) [reply libs/remix-ui/run-tab/src/lib/actions/deploy.ts:71]: I've added the "translation" label, for @ryestew

File: apps/remix-ide/src/app/tabs/locales/zh/udapp.json (hunk 1 above)
Comment (yann300) [reply apps/remix-ide/src/app/tabs/locales/zh/udapp.json:30]: I've added the "translation" label, for @ryestew.

**Review by ioedeveloper — APPROVED** [review]

**Pull request merged**